Classes/DDFileLogger.m

```diff
@@ -869,6 +874,32 @@ - (void)maybeRollLogFileDueToSize {
     }
 }
 
+- (void)maybeRollLogFileDueToDate {
+	if (_rollEveryDay && _currentLogFileInfo && ![self isLogFileCreatedToday:_currentLogFileInfo]) {
+		NSLogVerbose(@"DDFileLogger: Rolling log file due to date...");
+		
+		[self rollLogFileNow];
+	}
+}
+
+- (BOOL)isLogFileCreatedToday:(DDLogFileInfo *)aLogFile {
+	NSDate *logFileCreationDate = aLogFile.creationDate;
+	NSDate *today = [NSDate date];
```

Comment:
> With my flag this will not happen. in file "2016-05-23.log" will be messages only for 23, not for 24.

Depending on the time zone that is.

I think that if you need something very specific you could subclass the file logger and override the roll methods exactly to what you need. You will have all the flexibility to modify the behavior.